src/bridge-deployer/src/evm.rs

```diff
@@ -0,0 +1,31 @@
+//! EVM related utilities for link params.
+
+use bridge_did::evm_link::EvmLink;
+use candid::Principal;
+
+use crate::contracts::EvmNetwork;
+
+const MAINNET_PRINCIPAL: &str = "i3jjb-wqaaa-aaaaa-qadrq-cai";
+const TESTNET_PRINCIPAL: &str = "4fe7g-7iaaa-aaaak-aegcq-cai";
+
+/// Returns the IC host based on the EVM network.
+pub fn ic_host(evm_network: EvmNetwork) -> String {
+    match evm_network {
+        EvmNetwork::Localhost => "http://localhost:4943".to_string(),
```

Comment:
why port `4943`? dfx port is determined dynamically at startup. In this case, you should ask the user to provide the port number as a parameter or to detect the port dynamically (e.g. with the `dfx info replica-port` command).
Please also note that somewhere else in the deployer code, there is the `8545` hardcoded local port which has the same issue